cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java

```diff
@@ -141,16 +143,20 @@
     public static final String CONFIG_PROPERTY_BUCKETMODE = "bucketMode";
     public static final String CONFIG_PROPERTY_BUCKETDEPTH = "bucketLength";
     static final String CHECKSUM_ATTRIBUTE_NAME = "contentChecksum";
+    static final String CONTENT_FOLDER = "content";
+    static final String TXN_FOLDER = "transaction";
     
     static final String MD5_CHECKSUM_SCHEME = "md5";
     static final int MAX_BUCKET_LENGTH = 5;
     static final int DEFAULT_BUCKET_LENGTH = 2;
     
     private FileSystem fs;
     private Path root;
+    private Path contentFolder;
+    private Path transactionFolder;
     private BucketMode bucketMode;
     private int bucketLength = DEFAULT_BUCKET_LENGTH;
-    
+
```

Comment:
ok

Reply:
bucketMode can also be final. You can (should) move the assignment from init() to ctor(s).

Reply:
done

Reply:
just noticed that bucketLength is static instead of final. While you are at it - it should be mandatory with no default value for the mode that uses it (enforcing the MAX_BUCKET_LENGTH is fine -- 5 means 1e6 directories at level 5)